lms/djangoapps/courseware/views/views.py

```diff
@@ -440,6 +439,52 @@ def get(self, request, course_id, tab_type, **kwargs):
                 return CourseTabView.handle_exceptions(request, course, exception)
 
     @staticmethod
+    def url_to_enroll(course_key):
+        """
+        Returns the URL to use to enroll in the specified course.
+        """
+        # TODO: LEARNER-1697: Transition handling of enroll links in new Course Home.
+        # link to where the student should go to enroll in the course:
+        # about page if there is not marketing site, SITE_NAME if there is
+        url_to_enroll = reverse('about_course', args=[unicode(course_key)])
+        if settings.FEATURES.get('ENABLE_MKTG_SITE'):
+            url_to_enroll = marketing_link('COURSES')
+        return url_to_enroll
+
+    @staticmethod
+    def show_user_access_warnings(request, course_key):
+        """
+        Show messages to the user if they have limited access.
+        """
+        # Show messages to the user
+        is_enrolled = CourseEnrollment.is_enrolled(request.user, course_key)
+        is_staff = has_access(request.user, 'staff', course_key)
+        if request.user.is_anonymous():
+            register_warning_message(
+                request,
+                Text(_("To see course content, {sign_in_link} or {register_link}.")).format(
+                    sign_in_link=HTML('<a href="/login?next={current_url}">{sign_in_label}</a>').format(
+                        sign_in_label=_("Sign in"),
+                        current_url=urlquote_plus(request.path),
+                    ),
+                    register_link=HTML('<a href="/register?next={current_url}">{register_label}</a>').format(
+                        register_label=_("register"),
+                        current_url=urlquote_plus(request.path),
+                    ),
+                )
+            )
+        elif not is_enrolled and not is_staff:
+            register_warning_message(
+                request,
+                Text(_('You must be enrolled in the course to see course content. {enroll_button}')).format(
+                    enroll_button=HTML('<a href="{url_to_enroll}">{enroll_button_label}</a>').format(
+                        url_to_enroll=CourseTabView.url_to_enroll(course_key),
+                        enroll_button_label=_("Enroll Now"),
```

Comment:
@andy-armstrong @catong: The old message on the info page read: "Enroll now!".  I prefer the lowercase 'n'.  I also think we need punctuation (period or exclamation point).

I made the Performance course use the info page and unenrolled the honor student, so you can see this message here when logged in as honor:
https://andy-armstrong.sandbox.edx.org/courses/course-v1:DemoX+PERF101+course/